test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -855,6 +839,214 @@ UCS_TEST_P(test_uct_cm_sockaddr, connect_client_to_server_reject_with_delay)
+
+    void server_accept(entity *server, uct_conn_request_h conn_request,
+                       uct_ep_server_connect_cb_t connect_cb,
+                       uct_ep_disconnect_cb_t disconnect_cb,
+                       void *user_data)
+    {
+        server->accept(server->cm(), conn_request, connect_cb, disconnect_cb,
+                       user_data);
+    }
+
+    static void
+    conn_request_cb(uct_listener_h listener, void *arg,
+                    const char *local_dev_name,
+                    uct_conn_request_h conn_request,
+                    const uct_cm_remote_data_t *remote_data) {
+
+        test_uct_cm_sockaddr_stress *self =
+                        reinterpret_cast<test_uct_cm_sockaddr_stress *>(arg);
+
+        if (test_uct_cm_sockaddr::common_conn_request(listener, arg,
+                                                      local_dev_name,
+                                                      conn_request,
+                                                      remote_data)) {
+            self->server_accept(self->m_server, conn_request,
+                                server_connect_cb, server_disconnect_cb, self);
+        }
+    }
+
+protected:
+    int                     m_clients_num;
+    std::vector<ep_state_t> m_all_eps;
+    int                     m_ep_init_disconnect_cnt;
+    pthread_mutex_t         m_lock;
+};
+
+UCS_TEST_P(test_uct_cm_sockaddr_stress, many_clients_to_one_server)
+{
+    int i, no_disconnect_eps_cnt = 0;
+    entity *client_test;
+    time_t seed = time(0);
+    ucs_time_t deadline;
+
+    /* Listen */
+    start_listen(test_uct_cm_sockaddr_stress::conn_request_cb);
+
+    /* Connect */
+    /* multiple clients, each on a cm of its own, connecting to the same server */
+    for (i = 0; i < m_clients_num; ++i) {
+        client_test = uct_test::create_entity();
+        m_entities.push_back(client_test);
+
+        client_test->max_conn_priv = client_test->cm_attr().max_conn_priv;
+        client_test->connect(0, *m_server, 0, m_connect_addr, client_priv_data_cb,
+                             client_connect_cb, client_disconnect_cb, this);
+    }
+
+    /* wait for the server to connect to all the clients */
+    wait_for_client_server_counters(&m_server_connect_cb_cnt,
+                                    &m_client_connect_cb_cnt, m_clients_num);
+
+    EXPECT_EQ(m_clients_num, m_server_recv_req_cnt);
+    EXPECT_EQ(m_clients_num, m_client_connect_cb_cnt);
+    EXPECT_EQ(m_clients_num, m_server_connect_cb_cnt);
+    EXPECT_EQ(m_clients_num, (int)m_server->num_eps());
+
+    /* Disconnect */
+    srand(seed);
+    UCS_TEST_MESSAGE << "Using seed: " << seed;
+
+    pthread_mutex_init(&m_lock, NULL);
+    m_all_eps.resize(2 * m_clients_num);
+
+    /* save all the clients' and server's eps in the m_all_eps array */
+    for (i = 0; i < m_clients_num; ++i) {
+        /* first 2 entities are m_server and m_client */
+        m_all_eps[i].ep                    = m_entities.at(2 + i).ep(0);
```

Comment:
can we initialize this in the loop at line 946?

Reply:
in the loop in line 946 we can add only the eps of the client side since the server's ep are not created yet. also, we dont use the m_all_eps array there. 
isn't it better to initialize both client and server eps in one place right before we need to use them?

Reply:
ok